File: app/javascript/channels/answers_channel.js
import JsonpTemplatePlugin from "webpack/lib/web/JsonpTemplatePlugin";
import consumer from "./consumer";

consumer.subscriptions.create("AnswersChannel", {
  connected() {
    console.log("AnswersChannel connected");
    this.perform("follow", { question_id: gon.question_id });
  },

  disconnected() {
    console.log("AnswersChannel disconnected");
  },

  received(data) {
    if (data.answer.author_id === gon.current_user_id) {
      return;
    }
    if (!location.pathname.endsWith(data.answer.question_id)) {
      return;
    }
    data.disabled_vote = data.answer.author_id === gon.current_user_id;
    const answer = require("templates/answer.hbs")(data);
    document.querySelector(`.answers`).insertAdjacentHTML("beforeend", answer);
  },
});

File: app/channels/answers_channel.rb
# frozen_string_literal: true

class AnswersChannel < ApplicationCable::Channel
  def follow(data)
    Rails.logger.info data
    Rails.logger.info "AnswersChannel data: question_#{data['question_id']}_answers"
    stream_from "question_#{data['question_id']}_answers"
  end
end
